src/commands/server/restore.ts

```diff
@@ -85,10 +115,208 @@ export default class Restore extends Command {
   private getRestoreTasks(flags: any): Listr.ListrTask[] {
     return [
       {
-        title: 'Scheduling restore...',
+        title: 'Detecting existing operator version...',
+        enabled: () => flags.version || flags.rollback || flags['backup-cr'],
+        task: async (ctx: any, task: any) => {
+          const kube = new KubeHelper(flags)
+          let operatorDeploymentYaml = await kube.getDeployment(OPERATOR_DEPLOYMENT_NAME, flags.chenamespace)
+          if (!operatorDeploymentYaml) {
+            // There is no operator deployment in Che namespace
+            // Check if the operator is in all namespaces mode
+            operatorDeploymentYaml = await kube.getDeployment(OPERATOR_DEPLOYMENT_NAME, DEFAULT_OPENSHIFT_OPERATORS_NS_NAME)
+            if (!operatorDeploymentYaml) {
+              // Still no operator deployment found
+              ctx.currentOperatorVersion = ''
+              task.title = `${task.title} operator is not deployed`
+              return
+            }
+          }
+          const operatorEnv = operatorDeploymentYaml.spec!.template.spec!.containers[0].env!
+          const currentVersionEnvVar = operatorEnv.find(envVar => envVar.name === 'CHE_VERSION')
+          if (!currentVersionEnvVar) {
+            throw new Error(`Failed to find Che operator version in '${OPERATOR_DEPLOYMENT_NAME}' deployment in '${flags.chenamespace}' namespace`)
+          }
+          ctx.currentOperatorVersion = currentVersionEnvVar.value
+          task.title = `${task.title} ${ctx.currentOperatorVersion} found`
+        },
+      },
+      {
+        title: 'Detecting installer...',
+        // It is possible to skip '&& (flags.version || flags.rollback || flags['backup-cr'])' in the enabled condition below,
+        // as ctx.currentOperatorVersion is set only if previous task, that has the condition, is executed.
+        enabled: (ctx: any) => ctx.currentOperatorVersion && !flags['olm-channel'],
```

Comment:
We can always detect installer type.

Reply:
Not always. If operator is not deployed we cannot.